CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
+9.  The pull requests will be reviewed and merged by the Azure SDK team
 
 ### New Resource Provider
+
 1. If you have never created an SDK for your service before, you will need the following things to get your SDK in the repo
 2. Follow the standard process described above.
-3. Directory names helps in using basic heuristics in finding projects as well it's associated test projects during CI process.
-4. Create a new directory (name of your service e.g. Compute, Storage etc)
-6. For e.g. for management plane SDKs the following directory structure
- - `<RPName>\<packageName>\src\Microsoft.Azure.Management.<RPName>\Microsoft.Azure.Management.<RPName>.csproj`
- - `<RPName>\<packageName>\tests\Microsoft.Azure.Management.<RPName>.Tests.csproj`
+3. Project names helps in using basic heuristics in finding projects as well it's associated test projects during CI process.
+4. Create a new directory using the name of your service/ resource provider as specified in [azure-rest-api-specs/specification](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) Repo
+5. Follow the the directory structure below
+
+```
+sdk\<service name>\<package name>\README.md
+sdk\<service name>\<package name>\*src*
+sdk\<resource provider name>\<package name>\*tests*
+sdk\<resource provider name>\<package name>\*samples*
+```
+
+e.g.
+
+```
+sdk\eventgrid\Microsoft.Azure.EventGrid\src\Microsoft.Azure.EventGrid.csproj
+sdk\eventgrid\Microsoft.Azure.EventGrid\tests\Microsoft.Azure.EventGrid.Tests.csproj
+sdk\eventgrid\Microsoft.Azure.Management.EventGrid\src\Microsoft.Azure.Management.EventGrid.csproj
+sdk\eventgrid\Microsoft.Azure.Management.EventGrid\tests\Microsoft.Azure.Management.EventGrid.Tests.csproj
+```
+
+> \*Ensure that your service name is the same as it is specified in the [azure-rest-api-specs/specification](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) Repo, that your csproj files starts with **Microsoft.Azure\***
+> , that test files end with **.Tests** and that management plane project files containt **.Management.**
+
 7. Copy .csproj from any other .csproj and update the following information in the new .csproj
- - PackageId
- - Description
- - AssemblyTitle
- - AssemblyName
- - Version
- - PackageTags
- - PackageReleaseNotes (this is important because this information is displayed on www.nuget.org when your nuget package is published
-8. Copy existing generate.ps1 file from another dirctory and update the `ResourceProvider` name that is applicable to your SDK. Resource provider refers to the relative path of your REST spec directory in Azure-Rest-Api-Specs repository
-During SDK generation, this path helps to locate the REST API spec from the `https://github.com/Azure/azure-rest-api-specs`
+
+   | For DataPlane       | For Management Plane |
+   | ------------------- | -------------------- |
+   | AssemblyTitle       | PackageId            |
```

Comment:
I think we should normalize these. The managment plane doesn't need PackageId or AssemblyName as those are defaulted from the project. We should also start using VersionPrefix everywhere as it enables us to build the full version as needed.